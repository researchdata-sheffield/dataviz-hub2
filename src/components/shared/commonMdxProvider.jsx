import React from 'react'
import { MDXProvider } from "@mdx-js/react"
import { MDXRenderer } from "gatsby-plugin-mdx"
import { H1, H2, H3, H4, H5, H6, P, A, Ol, Li, Hr, Del, Pre, Ul, BlockQuote, Link, EM, Table, LPItem, LPWrap, IMG, TwitterBtn } from "../style/blogPostStyle"
import { Accordion, AccordionItem, AccordionItemHeading, AccordionItemButton, AccordionItemPanel } from 'react-accessible-accordion';
import Tab from "./tab";

export default function commonMdxProvider(props) {

  return (
    <MDXProvider 
      components={{ 
        h1: H1, h2: H2, h3: H3, h4: H4, h5: H5, h6: H6, p: P, a: A, ol: Ol, li: Li, hr: Hr, del: Del, 
        pre: Pre, ul: Ul, blockquote: BlockQuote, Link: Link, em: EM, img: IMG, table: Table, 
        Accordion, AccordionItem, AccordionItemHeading, AccordionItemButton, AccordionItemPanel, 
        LPWrap, LPItem, TwitterBtn, Tab, ...props.components
      }}
    >
      <MDXRenderer className={props.className} {...props}>{props.mdx.body}</MDXRenderer>
    </MDXProvider>
  )
}
