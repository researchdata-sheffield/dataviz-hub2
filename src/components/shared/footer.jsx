import { Link } from "gatsby"
import React from "react"
import university_logo from "../../images/TUOSlogo.png"
import { FaGoogle, FaSlack, FaRss } from "react-icons/fa"
import { MdEmail } from "react-icons/md"
import { A_footer } from "../style/styleComponent"

const Footer = () => {
  const currentYear = new Date();
  const linkClasses = "text-gray-400 hover:text-brand-blue"

  return (
    <footer className="bg-black px-8 pt-8 pb-4 lg:px-12 lg:pt-16 lg:pb-6 text-sm relative z-10">
      <div className="sm:flex mb-4">
        <div className="sm:w-2/12 h-auto">
            <div className="mb-2"><Link className="text-white font-bold text-md" to="/">Dataviz.Shef</Link></div>
            <ul className="list-reset leading-normal">
              <li><Link className={linkClasses} to="/blog">Blog</Link></li>
              <li><Link className={linkClasses} to="/events">Events</Link></li>
              <li><Link className={linkClasses} to="/community">Community</Link></li>
              <li><A_footer href="https://orda.shef.ac.uk/visualisations/">Showcase</A_footer></li>
              <li><Link className={linkClasses} to="/about">About</Link></li>
              <li><A_footer href="mailto:rdm@sheffield.ac.uk">Contact us</A_footer></li>
              <li><Link className={linkClasses} to="/privacy">Privacy Policy</Link></li>
              <li><Link className={linkClasses} to="/accessibility">Accessibility</Link></li>
              <li><Link className={linkClasses} to="/sitemap/sitemap-index.xml">Sitemap</Link></li>
            </ul>
        </div>
        <div className="sm:w-3/12 h-auto sm:mt-0 mt-8">
            <div className="text-md mb-2 font-bold text-white">Online Research Data</div>
              <ul className="list-reset leading-normal">
                <li><A_footer href="https://orda.shef.ac.uk/articles/list/desc/published_date/all">Datasets</A_footer></li>
                <li><A_footer href="https://orda.shef.ac.uk/#orda-fac">Faculties</A_footer></li>
                <li><A_footer href="https://figshare.shef.ac.uk/collections/University_of_Sheffield_visualisation_showcase/3879643">Visualised datasets</A_footer></li>
              </ul>

          <div className="mb-2 mt-4 text-md font-bold text-white">The University of Sheffield</div>
            <ul className="list-reset leading-normal">
            <li><A_footer href="https://sheffield.ac.uk/library/rdm">Research Data Management support</A_footer></li>
                <li><A_footer href="https://www.sheffield.ac.uk/it-services">IT Services</A_footer></li>
                <li><A_footer href="https://rse.shef.ac.uk/">Research Software Engineering</A_footer></li>
            </ul>

        </div>
        <div className="sm:w-2/12 h-auto sm:mt-0 mt-8">
          <div className="text-white text-md mb-2 font-bold">Collaboration</div>
          <ul className="list-reset leading-normal">
            <li><A_footer href="https://github.com/researchdata-sheffield/dataviz-hub2">Github repository</A_footer></li>
            <li><Link className={linkClasses} to="/docs/22/03/2020/contribute-blog-post">Contribute blog post</Link></li>
          </ul>
        </div>
        
        <div className="sm:w-5/12 sm:flex sm:mt-0 mt-8 h-auto justify-center flex-wrap">
          <div className="sm:w-2/3 text-white text-lg mb-2 pr-6">
            <div className="font-semibold text-brand-blue">Data Visualisation Hub </div>
            <div className="text-gray-100 text-lg mb-2 font-medium">The University of Sheffield.</div>
            <p className="text-gray-400 leading-normal text-sm">
              To help our researchers make the most of their data and take advantage of tools, we have been working 
              on Dataviz.Shef, a multi-pronged initiative to provide tools, training and build a community around 
              interactive data visualisation at TUoS.
            </p>
          </div>
            
          <div className="sm:w-1/3 pl-6">
            <A_footer className="m-20" href="https://www.sheffield.ac.uk/" title="The University of Sheffield Logo">
              <img className="w-11/12" src={university_logo} alt="The University of Sheffield Logo"></img>
            </A_footer>
            <div className="flex flex-wrap text-center justify-around w-11/12 text-gray-300">
              <A_footer href="https://groups.google.com/a/sheffield.ac.uk/forum/?hl=en#!forum/shef_dataviz-group" title="Join Google group"><FaGoogle /></A_footer>
              <A_footer href="https://join.slack.com/t/shef-dataviz/signup" title="Join Slack Channel"><FaSlack /></A_footer>
              <A_footer title="RSS Feed" target="_self" href="/rss.xml"><FaRss /></A_footer>
              <A_footer href="mailto:rdm@sheffield.ac.uk" title="contact us on email"><MdEmail /></A_footer>
            </div>
          </div>
        </div>
      </div>
      <p className="text-gray-500 leading-normal text-center text-sm pt-16">
        Â© 2018 - {currentYear.getFullYear()} The University of Sheffield.
      </p>
    </footer>
  )
}

export default Footer
