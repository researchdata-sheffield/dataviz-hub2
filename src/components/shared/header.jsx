import { Link } from "gatsby";
import PropTypes from "prop-types";
import React, { useState, useEffect } from "react";
import university_logo from "../../images/TUOSlogo.png";
import { FiSearch } from "react-icons/fi";
import { FaGoogle, FaSlack, FaTimes, FaBars, FaRss } from "react-icons/fa";
import { A } from "../style/blogPostStyle";
import { useLocation } from "@gatsbyjs/reach-router";

const Header = () => {
  const [isExpanded, toggleExpansion] = useState(false);
  const [isScroll, toggleScrolled] = useState(false);
  const [isOpen, toggleOpen] = useState(false);
  const location = useLocation();
  var currentPagePath = location.pathname;

  // close mobile menu on route change
  if (typeof window !== "undefined" && window.screen.width <= 1200) {
    useEffect(() => {
      toggleExpansion(false);
    }, [location]);
  }

  // monitor page scroll
  useEffect(() => {
    function handleScroll() {
      const scrolledValue = window.scrollY > 10;
      if (isScroll !== scrolledValue) {
        toggleScrolled(!isScroll);
      }
    }
    document.addEventListener("scroll", handleScroll, { passive: true });
    return () => {
      document?.removeEventListener("scroll", handleScroll);
    };
  }, [isScroll]);

  // RSS button
  useEffect(() => {
    document.querySelector("#copyBtn").addEventListener("click", copyRssLink);
    document.querySelector("#rssMobile").addEventListener("click", copyRssLink);
    document
      .querySelector("#rssDesktop")
      .addEventListener("click", copyRssLink);

    return () => {
      document
        ?.querySelector("#copyBtn")
        ?.removeEventListener("click", copyRssLink);
      document
        ?.querySelector("#rssMobile")
        ?.removeEventListener("click", copyRssLink);
      document
        ?.querySelector("#rssDesktop")
        ?.removeEventListener("click", copyRssLink);
    };
  });

  onScrollNav();

  /**
   * Sticky nav. Hide on scroll down, visible on scroll up
   */
  function onScrollNav() {
    if (typeof window === "undefined") {
      return;
    }

    let prevScrollPos = window.pageYOffset;

    window.onscroll = function () {
      const maxScroll = document.body.clientHeight - window.innerHeight;
      let currentScrollPos = window.pageYOffset;
      if (
        (maxScroll > 0 &&
          prevScrollPos > currentScrollPos &&
          prevScrollPos <= maxScroll) ||
        (maxScroll <= 0 && prevScrollPos > currentScrollPos) ||
        (prevScrollPos <= 0 && currentScrollPos <= 0)
      ) {
        document.getElementById("navbar").style.top = "0px";
        document.getElementById("navbar").style.visibility = "visible";
      } else {
        document.getElementById("navbar").style.top = "-300px";
        document.getElementById("navbar").style.paddingTop = "1px";
        document.getElementById("navbar").style.paddingBottom = "1px";
        document.getElementById("navbar").style.visibility = "hidden";
      }
      prevScrollPos = currentScrollPos;
    };
  }

  /**
   * Return className for nav links
   * @param {*} props
   * @param {string} props.type: Link / a
   * @param {string} props.className
   */
  const NavLink = (props) => {
    function navColourClass(mobile = false) {
      var className = "transition duration-300 relative "; // don't remove space

      if (mobile == "true") {
        if (isScroll) {
          className += "text-black hover:text-brand-blue";
          return className;
        }
        className += "text-gray-600 hover:text-brand-blue";
        return className;
      }

      if (isScroll) {
        className += "text-black hover:text-brand-blue";
        return className;
      }

      className += "text-gray-100 group-hover:text-black hover:text-brand-blue";
      return className;
    }

    switch (props.type) {
      case "a":
        return (
          <A
            {...props}
            className={`${navColourClass(props.mobile)} ${
              props.className
            } transition duration-500 ease-in-out md:mr-2 no-underline px-4 py-1`}
            style={{ background: "none" }}
          >
            {props.children}
          </A>
        );

      case "Link":
        return (
          <Link
            to={props.to}
            {...props}
            className={`${navColourClass(props.mobile)} ${
              props.className
            } transition duration-500 ease-in-out md:mr-2 no-underline px-4 py-1`}
          >
            {props.children}
          </Link>
        );

      case "a-icon":
        return (
          <A
            {...props}
            className={`${navColourClass(props.mobile)} ${
              props.className
            } flex items-center`}
            style={{ background: "none" }}
          >
            {props.children}
          </A>
        );

      case "Link-icon":
        return (
          <Link
            to={props.to}
            {...props}
            className={`${navColourClass(props.mobile)} ${
              props.className
            } flex items-center`}
          >
            {props.children}
          </Link>
        );
    }
  };

  function mobileMenuIconClass() {
    var className =
      "px-3 py-2 flex items-center border rounded outline-none transition duration-300 group-hover:border-gray-700 group-hover:text-gray-700";

    if (isExpanded || isScroll || currentPagePath === "/") {
      className += " text-gray-700 border-gray-700"; // do not delete the space
      return className;
    }
    className += " text-white border-white"; // do not delete the space
    return className;
  }

  function copyRssLink() {
    toggleOpen(true);
    var copyText = document.querySelector("#rssLink");
    copyText.select();
    copyText.setSelectionRange(0, 9999);
    document.execCommand("copy");
  }

  return (
    <header className="font-semibold z-50 relative group">
      <nav
        id="navbar"
        role="navigation"
        className={`${
          isScroll
            ? "shadow-lg bg-white"
            : "group-hover:bg-white bg-transparent"
        } transition duration-500 flex items-center justify-between flex-wrap px-5 fixed w-full z-10 overflow-hidden`}
        style={{ transition: "top 0.3s" }}
      >
        <div className="flex items-center shrink-0 mr-5">
          <A
            className={`${
              isScroll ? "" : [currentPagePath === "/" ? "invisible" : ""]
            } `}
            href="https://www.sheffield.ac.uk/"
            title="The University of Sheffield Logo"
          >
            <img
              className="mt-1"
              alt="The University of Sheffield Logo"
              style={{ maxWidth: "13.6vh" }}
              src={university_logo}
            />
          </A>
          <div
            className={`${
              isScroll ? "" : [currentPagePath === "/" ? "invisible" : ""]
            } ml-4 text-lg font-extrabold transition duration-1000 ease-in-out`}
          >
            <Link className="textanimate" to="/">
              Dataviz.Shef
            </Link>
          </div>
        </div>

        {/* Mobile menu icon */}
        <div
          onClick={() => toggleExpansion(!isExpanded)}
          className="block xl:hidden py-2"
          style={{ zIndex: "999" }}
        >
          <button className={mobileMenuIconClass()} title="Menu">
            {isExpanded ? <FaTimes /> : <FaBars />}
          </button>
        </div>

        {/* Mobile devices */}
        <div
          id="mobileHeader"
          className={`${
            isExpanded
              ? "visible"
              : "invisible opacity-0 translate-x-full transform"
          } rounded-md xl:hidden shadow-lg px-6 py-10 md:py-16 fixed top-0 left-0 h-full min-w-full overflow-y-scroll`}
          style={{
            zIndex: "990",
            transition: ".3s linear",
            backgroundColor: "#fbfbfb"
          }}
        >
          {/* <div className="mt-5 pt-6 border-gray-200 border-t-1"></div> */}
          <div className="mt-5 justify-end flex flex-wrap items-center">
            {[
              {
                route: `/`,
                title: `Home`
              },
              {
                route: `/community`,
                title: `Community`
              },
              {
                route: `/events`,
                title: `Events`
              },
              {
                route: `/blog`,
                title: `Blog`
              },
              {
                route: `/visualisation`,
                title: `Visualisation`
              },
              {
                route: `https://orda.shef.ac.uk/visualisations/`,
                title: `Showcase`
              },
              {
                route: `/about`,
                title: `About`
              }
            ].map((link) =>
              link.title != "Showcase" ? (
                <NavLink
                  type="Link"
                  mobile="true"
                  className="block w-full text-2xl"
                  activeStyle={{
                    color: `${isScroll ? "#00aeef" : "black"}`,
                    fontWeight: "bold"
                  }}
                  key={link.title}
                  to={link.route}
                >
                  {link.title}
                  {link.title == "Visualisation" && (
                    <span
                      className="absolute top-0 right-0 bg-red-500 text-white rounded-md text-xs -mr-2"
                      style={{ padding: "2px 5px" }}
                    >
                      new
                    </span>
                  )}
                </NavLink>
              ) : (
                <NavLink
                  type="a"
                  mobile="true"
                  className="block w-full text-2xl"
                  href={link.route}
                  key={link.title}
                >
                  {link.title}
                </NavLink>
              )
            )}
            <div className="flex flex-wrap w-full justify-center space-x-4 mt-8 pt-6 border-t-2 border-gray-200">
              <NavLink
                type="a-icon"
                mobile="true"
                className="block mt-3"
                href="https://groups.google.com/a/sheffield.ac.uk/g/shef_dataviz-group"
                title="Join Google group"
              >
                <div className="px-4 text-2xl transition duration-500 ease-in-out no-underline">
                  <FaGoogle />
                </div>
              </NavLink>
              <NavLink
                type="a-icon"
                mobile="true"
                className="block mt-3"
                href="https://shef-dataviz.slack.com/join/signup#/domain-signup"
                title="Join Slack channel"
              >
                <div className="px-4 text-2xl transition duration-500 ease-in-out no-underline">
                  <FaSlack />
                </div>
              </NavLink>
              <NavLink
                id="rssMobile"
                type="Link-icon"
                mobile="true"
                className="block mt-3"
                target="_self"
                title="RSS Feed"
                onClick={() => {
                  return false;
                }}
              >
                <button className="relative px-4 text-2xl transition duration-500 ease-in-out no-underline">
                  <FaRss />
                </button>
              </NavLink>
              <NavLink
                type="Link-icon"
                mobile="true"
                className="block mt-3"
                to="/search"
                title="search dataviz hub"
              >
                <div className="px-4 text-2xl transition duration-500 ease-in-out no-underline">
                  <FiSearch />
                </div>
              </NavLink>
            </div>
          </div>
        </div>

        {/* Desktop Nav items // backgroundColor: `${isExpanded ? [isScroll ? '' : "rgba(25,25,25,.9)" ] : ""}`  */}
        <div
          id="desktopHeader"
          className={`hidden py-2 xl:flex items-center w-auto text-xs xl:text-sm`}
        >
          <div className="justify-end flex flex-wrap items-center">
            {[
              {
                route: `/community`,
                title: `Community`
              },
              {
                route: `/events`,
                title: `Events`
              },
              {
                route: `/blog`,
                title: `Blog`
              },
              {
                route: `/visualisation`,
                title: `Visualisation`
              },
              {
                route: `https://orda.shef.ac.uk/visualisations/`,
                title: `Showcase`
              },
              {
                route: `/about`,
                title: `About`
              }
            ].map((link) =>
              link.title != "Showcase" ? (
                <NavLink
                  type="Link"
                  activeStyle={{ color: `#00aeef`, fontWeight: "bold" }}
                  partiallyActive={true}
                  key={link.title}
                  to={link.route}
                >
                  {link.title}
                  {link.title == "Visualisation" && (
                    <span
                      className="invisible group-hover:visible absolute top-0 right-0 bg-red-500 text-white rounded-md -mr-3"
                      style={{
                        padding: "1px 3px",
                        fontSize: ".68rem",
                        lineHeight: "1rem"
                      }}
                    >
                      new
                    </span>
                  )}
                </NavLink>
              ) : (
                <NavLink type="a" href={link.route} key={link.title}>
                  {link.title}
                </NavLink>
              )
            )}
            <NavLink
              type="a-icon"
              href="https://groups.google.com/a/sheffield.ac.uk/g/shef_dataviz-group"
              title="Join Google group"
            >
              <div className="text-md transition duration-500 ease-in-out inline-block no-underline ml-2 mr-2 -mb-1">
                <FaGoogle />
              </div>
            </NavLink>
            <NavLink
              type="a-icon"
              href="https://shef-dataviz.slack.com/join/signup#/domain-signup"
              title="Join Slack channel"
            >
              <div className="text-md transition duration-500 ease-in-out inline-block no-underline ml-2 mr-2 -mb-1">
                <FaSlack />
              </div>
            </NavLink>
            <NavLink
              id="rssDesktop"
              type="a-icon"
              title="RSS Feed"
              target="_self"
              onClick={() => {
                return false;
              }}
            >
              <button className="relative text-md transition duration-500 ease-in-out inline-block no-underline ml-2 mr-2 -mb-1 cursor-pointer">
                <FaRss />
              </button>
            </NavLink>
            <NavLink type="Link-icon" to="/search" title="Search">
              <div className="mr-4 ml-2 text-sm xl:text-lg transition duration-500 ease-in-out md:mr-2 inline-block no-underline">
                <FiSearch />
              </div>
            </NavLink>
          </div>
        </div>

        {/* RSS BOX */}
        <div
          id="rssBox"
          className={`${
            isOpen ? "" : "invisible"
          } fixed bg-white shadow-2xl rounded-lg p-5 lg:p-10 text-center`}
          style={{
            top: "50%",
            left: "50%",
            transform: "translate(-50%, -50%)",
            zIndex: "1000"
          }}
        >
          <FaTimes
            className="text-2xl absolute top-0 right-0 text-gray-800 m-2 rounded-full hover:text-white hover:bg-red-500 transition duration-300 p-1"
            onClick={() => {
              toggleOpen(!isOpen);
            }}
          />
          <FaRss className="text-5xl mx-auto" />
          <h2 className="text-2xl font-bold mt-4">RSS Feed link copied!</h2>
          <h6 className="text-gray-600 text-sm mt-1">
            Doesn&apos;t work? Use the following link
          </h6>
          <div className="inline-block bg-white shadow rounded-md mt-5">
            <input
              id="rssLink"
              readOnly={true}
              className="py-2 pl-2 text-base focus:outline-none pr-3 text-gray-500"
              type="text"
              value="https://dataviz.shef.ac.uk/rss.xml"
            />
            <button
              id="copyBtn"
              className="bg-brand-pink p-2 rounded-md cursor-pointer"
              onClick={() => copyRssLink}
            >
              &nbsp; COPY&nbsp;
            </button>
          </div>
        </div>
      </nav>
    </header>
  );
};

Header.propTypes = {
  siteTitle: PropTypes.string,
  type: PropTypes.string,
  children: PropTypes.any,
  location: PropTypes.any,
  className: PropTypes.string,
  mobile: PropTypes.bool,
  to: PropTypes.any
};

Header.defaultProps = {
  siteTitle: ``
};

export default Header;
