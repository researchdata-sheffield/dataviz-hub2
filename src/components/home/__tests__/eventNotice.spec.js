/**
 * @jest-environment jsdom
 */
/* eslint-disable no-unused-vars */
/* eslint-disable no-undef */
import React from "react";
import { render, getByText } from "@testing-library/react";
import EventNotice from "../eventNotice";
import moment from "moment-timezone";
import { calculateUserLocalTime } from "../../../utils/shared";

jest.mock("../../../utils/shared", () => {
  return {
    ...jest.requireActual("../../../utils/shared"),
    calculateUserLocalTime: (timeNow) => {
      return {
        time: timeNow,
        timezone: "Europe/London"
      };
    }
  };
});

describe("Home | Event Notice Banner", () => {
  const mockData = {
    edges: [
      {
        node: {
          id: "163679197887",
          url: "https://www.eventbrite.co.uk/e/celebration-of-open-research-at-the-university-of-sheffield-tickets-163679197887",
          name: {
            text: "Celebration of Open Research at The University of Sheffield"
          },
          logo: {
            original: {
              url: "https://img.evbuc.com/https%3A%2F%2Fcdn.evbuc.com%2Fimages%2F143589123%2F121727010915%2F1%2Foriginal.20210804-163431?auto=format%2Ccompress&q=75&sharp=10&s=c61552c6c69fe01890ae8e6316ae1c2b"
            }
          },
          venue: null,
          online_event: true,
          summary:
            "Join us for a celebration of Open Research at the University of Sheffield with the winners of the Library's inaugural Open Research Prize.",
          start: {
            local: "Wed 27 October 2021, 10:00 AM"
          }
        }
      }
    ]
  };

  beforeAll(() => {
    jest.spyOn(console, "warn").mockImplementation(() => {});
    Date.now = jest.fn(() => new Date("2021-09-16T10:00:00.00"));
  });

  let notice;

  beforeEach(async () => {
    notice = render(<EventNotice eventBrite={mockData} />);
  });
  it("should renders correctly", async () => {
    expect(notice).toMatchSnapshot();
  });

  it("should display correct date and timezone", async () => {
    let timeNow = moment().format("ddd DD MMMM YYYY, hh:mm A");
    let userTimezone = calculateUserLocalTime(timeNow).timezone;

    expect(
      await getByText(notice.container, `${timeNow} (${userTimezone})`)
    ).toBeInTheDocument();
  });
});
