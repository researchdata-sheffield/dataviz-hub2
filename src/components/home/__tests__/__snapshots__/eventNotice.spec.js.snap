// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home | Event Notice Banner should renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="bg-black w-full"
      >
        <div
          class="react-reveal p-2 w-full text-gray-800 flex-col flex-wrap overflow-auto shadow-md border-t-1 border-gray-100 relative bg-white"
          id="eventNotice"
        >
          <div>
            <div
              class="text-gray-700 text-sm w-full font-bold p-3"
            >
              Thu 16 September 2021, 10:00 AM
               (
              Europe/London
              )
            </div>
            <a
              class="flex flex-wrap w-full overflow-hidden max-h-90 text-gray-700 group pb-2 px-2"
              href="https://www.eventbrite.co.uk/e/celebration-of-open-research-at-the-university-of-sheffield-tickets-163679197887"
              rel="noopener noreferrer"
              style="font-family: TUoS Blake;"
              target="_blank"
            >
              <img
                alt="Thumbnail: Join us for a celebration of Open Research at the University of Sheffield with the winners of the Library's inaugural Open Research Prize."
                class="w-full md:w-3/12 overflow-hidden self-center"
                src="https://img.evbuc.com/https%3A%2F%2Fcdn.evbuc.com%2Fimages%2F143589123%2F121727010915%2F1%2Foriginal.20210804-163431?auto=format%2Ccompress&q=75&sharp=10&s=c61552c6c69fe01890ae8e6316ae1c2b"
                style="object-fit: cover; object-position: center; min-height: 180px;"
              />
              <div
                class="w-full md:w-9/12 pb-2 px-4"
              >
                <h1
                  class="inline-block font-semibold xl:text-xl"
                >
                  <p
                    class="text-brand-blue text-2xl xl:text-3xl inline-block"
                  >
                    NEXT
                  </p>
                   
                  event:  
                  <p
                    class="inline-block text-lg xl:text-xl font-bold"
                  >
                    Celebration of Open Research at The University of Sheffield
                  </p>
                </h1>
                <p
                  class="text-gray-500 hidden md:flex lg:flex xl:flex leading-tight text-sm xl:text-base"
                >
                  Join us for a celebration of Open Research at the University of Sheffield with the winners of the Library's inaugural Open Research Prize.
                </p>
                <p
                  class="flex pt-2 text-sm xl:text-base"
                >
                  <svg
                    class="mr-1"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M256,8C119,8,8,119,8,256S119,504,256,504,504,393,504,256,393,8,256,8Zm92.49,313h0l-20,25a16,16,0,0,1-22.49,2.5h0l-67-49.72a40,40,0,0,1-15-31.23V112a16,16,0,0,1,16-16h32a16,16,0,0,1,16,16V256l58,42.5A16,16,0,0,1,348.49,321Z"
                    />
                  </svg>
                  Wed 27 October 2021, 10:00 AM
                   (
                  Europe/London
                  )
                </p>
                <div
                  class="flex flex-wrap"
                >
                  <div
                    class="w-full md:w-5/6 text-sm xl:text-base"
                  >
                    <p
                      class="flex"
                    >
                      <svg
                        class="mr-1 mt-1"
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 384 512"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z"
                        />
                      </svg>
                       
                      Online Event
                    </p>
                    <p
                      class="flex"
                    />
                  </div>
                  <div
                    class="flex flex-wrap w-full md:w-1/6 content-center justify-center hidden-xs hidden-md"
                  >
                    <button
                      class="hidden group-hover:flex shadow-sm -mt-4 py-2 px-5 text-lg bg-gray-800 text-white hover:bg-brand-blue"
                      href="https://www.eventbrite.co.uk/e/celebration-of-open-research-at-the-university-of-sheffield-tickets-163679197887"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      Register
                    </button>
                  </div>
                </div>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="bg-black w-full"
    >
      <div
        class="react-reveal p-2 w-full text-gray-800 flex-col flex-wrap overflow-auto shadow-md border-t-1 border-gray-100 relative bg-white"
        id="eventNotice"
      >
        <div>
          <div
            class="text-gray-700 text-sm w-full font-bold p-3"
          >
            Thu 16 September 2021, 10:00 AM
             (
            Europe/London
            )
          </div>
          <a
            class="flex flex-wrap w-full overflow-hidden max-h-90 text-gray-700 group pb-2 px-2"
            href="https://www.eventbrite.co.uk/e/celebration-of-open-research-at-the-university-of-sheffield-tickets-163679197887"
            rel="noopener noreferrer"
            style="font-family: TUoS Blake;"
            target="_blank"
          >
            <img
              alt="Thumbnail: Join us for a celebration of Open Research at the University of Sheffield with the winners of the Library's inaugural Open Research Prize."
              class="w-full md:w-3/12 overflow-hidden self-center"
              src="https://img.evbuc.com/https%3A%2F%2Fcdn.evbuc.com%2Fimages%2F143589123%2F121727010915%2F1%2Foriginal.20210804-163431?auto=format%2Ccompress&q=75&sharp=10&s=c61552c6c69fe01890ae8e6316ae1c2b"
              style="object-fit: cover; object-position: center; min-height: 180px;"
            />
            <div
              class="w-full md:w-9/12 pb-2 px-4"
            >
              <h1
                class="inline-block font-semibold xl:text-xl"
              >
                <p
                  class="text-brand-blue text-2xl xl:text-3xl inline-block"
                >
                  NEXT
                </p>
                 
                event:  
                <p
                  class="inline-block text-lg xl:text-xl font-bold"
                >
                  Celebration of Open Research at The University of Sheffield
                </p>
              </h1>
              <p
                class="text-gray-500 hidden md:flex lg:flex xl:flex leading-tight text-sm xl:text-base"
              >
                Join us for a celebration of Open Research at the University of Sheffield with the winners of the Library's inaugural Open Research Prize.
              </p>
              <p
                class="flex pt-2 text-sm xl:text-base"
              >
                <svg
                  class="mr-1"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M256,8C119,8,8,119,8,256S119,504,256,504,504,393,504,256,393,8,256,8Zm92.49,313h0l-20,25a16,16,0,0,1-22.49,2.5h0l-67-49.72a40,40,0,0,1-15-31.23V112a16,16,0,0,1,16-16h32a16,16,0,0,1,16,16V256l58,42.5A16,16,0,0,1,348.49,321Z"
                  />
                </svg>
                Wed 27 October 2021, 10:00 AM
                 (
                Europe/London
                )
              </p>
              <div
                class="flex flex-wrap"
              >
                <div
                  class="w-full md:w-5/6 text-sm xl:text-base"
                >
                  <p
                    class="flex"
                  >
                    <svg
                      class="mr-1 mt-1"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 384 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z"
                      />
                    </svg>
                     
                    Online Event
                  </p>
                  <p
                    class="flex"
                  />
                </div>
                <div
                  class="flex flex-wrap w-full md:w-1/6 content-center justify-center hidden-xs hidden-md"
                >
                  <button
                    class="hidden group-hover:flex shadow-sm -mt-4 py-2 px-5 text-lg bg-gray-800 text-white hover:bg-brand-blue"
                    href="https://www.eventbrite.co.uk/e/celebration-of-open-research-at-the-university-of-sheffield-tickets-163679197887"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    Register
                  </button>
                </div>
              </div>
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
